tools/ocean/argo_decoder.xml

```diff
@@ -0,0 +1,63 @@
+<tool id="coriolis_data_processing_chain_for_argo_floats" name="coriolis data processing" version="@VERSION@">
+    <description>chain for argo floats</description>
+    <macros>
+        <token name="@VERSION@">061a</token>
+        <token name="@VERSION_DOCKER@">@VERSION@-galaxy</token>
+    </macros>
+    <requirements>
+        <container type="docker">ghcr.io/euroargodev/coriolis-data-processing-chain-for-argo-floats:@VERSION_DOCKER@</container>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        
+        ## prepare outputs directory
+        mkdir -p /mnt/data/output/log &&
+        chmod -R 770 /mnt/data/output &&
+
+        ## load inputs into data directory
+        cp '$raw_data' '/tmp/raw_data.zip' &&
+        unzip /tmp/raw_data.zip -d /mnt/data &&
+
+        ## execute decoder script
+        ##$APP_HOME/$APP_RUN_FILE /mnt/runtime 'rsynclog' 'all' 'configfile' '/app/config/_argo_decoder_conf_ir_sbd.json' 'xmlreport' 'argo-decoder-report.xml' 'floatwmo' '$float_wmo' 'PROCESS_REMAINING_BUFFERS' '1' &&
+        /app/run_decode_argo_2_nc_rt.sh /mnt/runtime 'rsynclog' 'all' 'configfile' '/app/config/_argo_decoder_conf_ir_sbd.json' 'xmlreport' 'argo-decoder-report.xml' 'floatwmo' '$float_wmo' 'PROCESS_REMAINING_BUFFERS' '1' &&
+
+        ## copy (because we need to be in working directory) and create output archive
+        cp -R /mnt/data/output/* . &&
+        zip -r argo-decoder-outputs.zip .
+
+    ]]>
+    </command>
+     <environment_variables>
```

Comment:
This is not needed anymore I assume? Its not used in the command line.

Reply:
yes we are still working on understanding how the environment var works here because when we defined them here it does not seem to know them.

Reply:
Maybe you ENV are not available for all users and the user which Galaxy is starting the container has no access to them?

Reply:
The environments variables are declared directly in the Dockerfile :

```Dockerfile
ENV APP_DIR=/app
ENV APP_RUN_FILE=run_decode_argo_2_nc_rt.sh
```

I tried to run the container with a random user on my computer :

```bash
docker run -it -u 123456:123456 ghcr.io/euroargodev/coriolis-data-processing-chain-for-argo-floats:061a-galaxy sh
$ id   
uid=123456 gid=123456 groups=123456
$ echo $APP_HOME/$APP_RUN_FILE             
/app/run_decode_argo_2_nc_rt.sh
$ printenv
HOSTNAME=307c9416801e
HOME=/
DATA_HOME=/mnt/data
MCR_CACHE_ROOT=/tmp/matlab/cache
REF_HOME=/mnt/ref
TERM=xterm
APP_HOME=/app
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
APP_DIR=/app
RUNTIME_HOME=/mnt/runtime
PWD=/app
APP_RUN_FILE=run_decode_argo_2_nc_rt.sh
```

Is it possible that it's a synthaxe problem in our configuration file ? I have the impression that when we use the `$` character there's an error even before the script runs.

Reply:
@bgruening would you know what's the issue here ?

Reply:
If you use a $env then you need to shield the env. Because in the template language the $ is a reserved keyword. Use \$env.